### PR TITLE
Add search and filtering to resources table

### PR DESCRIPTION
**Pull request by kewitham**
[Resource Table Search Implementation #1057](https://github.com/usdoj-crt/beta.ADA.gov-Project-Management/issues/1057)

## What does this change?
This PR adds an api path to facilitate dynamic searching and filtering on the resources table.

## Screenshots (for front-end PR):
Image

### Author

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for, document, and establish a testing plan for any behavior that may vary across environments or is otherwise difficult to test.
+ [ ] Check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

### Reviewer

+ [ ] If this is a story, run locally and check to make sure all Acceptance Criteria are met. (If any criteria are unclear, ask about them.).
+ [ ] Check for any behavior that may vary across environments or is difficult to test, and ensure that it is well-understood, documented, and that there is a testing plan in place.
+ [ ] Re-check for [accessibility](https://github.com/usdoj-crt/crt-portal/blob/develop/docs/a11y_plan.md).
+ [ ] [Tests pass](https://github.com/USDOJ/crt-portal/#tests).

## Notes for reviewer:

See [PR instructions doc](https://github.com/usdoj/crt-portal/blob/master/docs/pull_requests.md) for full pull request review instructions.

### CODE REVIEW

File: crt_portal/static/js/resources.js [hunk 1]
```diff
@@ -91,21 +118,154 @@
...
+      setTimeout(() => {
+        makeQuery();
+      }, 1500);
     };
-  });
+
+    const tagInput = document.querySelector('#id_tags-assign-tag');
+    tagInput.onchange = function() {
+      makeQuery();
+    };
```
Comment (okkays) [inline crt_portal/static/js/resources.js:151]: you may want to cancel previous setTimeouts so things don't come back out of order.

https://stackoverflow.com/questions/452003/cancel-kill-window-settimeout-before-it-happens

**Review by okkays — CHANGES_REQUESTED** [review]
really nice work! One suggestion, but this pattern is going to be a _huge_ improvement to portal UX :D

File: crt_portal/static/js/resources.js (hunk 1 above)
Comment (okkays) [reply crt_portal/static/js/resources.js]: @kewitham I sort of expected `clearTimeout` to be called - am I missing something?

Comment (kewitham) [reply crt_portal/static/js/resources.js]: Yep I added clearTimeout for every time setTimeout is used. There is no timeout for the tag input event listener.

Comment (okkays) [reply crt_portal/static/js/resources.js]: ah mb - must have been looking at the outdated version - thanks for putting up with me :D

**Review by okkays — APPROVED** [review]

**Pull request merged**